src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -68,9 +68,9 @@ private static void BindProperty(PropertyInfo property, object propertyOwner, IC
             }
         }
 
-        private static object BindType(Type type, object typeInstance, IConfiguration configuration)
+        private static object BindType(Type type, object typeInstance, IConfigurationSection configuration)
         {
-            var configValue = configuration.Get(null);
+            var configValue = configuration.Value;
```

Comment:
@HaoK Yes, I arrived to that when I looked the second time. It is only that in our current design a config root will never have a value now, so only need to look for the value if it is a section.